Assets/Scripts/UI/HealthEffectController.cs

```diff
@@ -0,0 +1,134 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Rendering;
+using UnityEngine.Rendering.Universal;
+
+public class HealthEffectController : MonoBehaviour
+{
+   
+   public Volume volume;
+
+   private Vignette vignette;
+   public bool vignetteToggle;
+
+   private FilmGrain filmgrain;
+   public bool filmgrainToggle;
+
+   private PaniniProjection paniniprojection;
+   public bool distanceToggle;
+   public bool cropToFitToggle;
+
+   private LensDistortion lensdistortion;
+   public bool lensDistortionToggle;
+
+   private Bloom bloom;
+   public bool bloomToggle;
+
+   public Killable killable;
+
+   void Start()
+   {
+
+    if(volume.profile.TryGet<Vignette>(out vignette))
+    {
+        killable.OnHealthChange.AddListener(OnHealthChange);
+    }
+
+    if(volume.profile.TryGet<FilmGrain>(out filmgrain))
+    {
+        killable.OnHealthChange.AddListener(OnHealthChange);
+    }
+
+    if(volume.profile.TryGet<PaniniProjection>(out paniniprojection))
+    {
+        killable.OnHealthChange.AddListener(OnHealthChange);
+    }
+
+    if(volume.profile.TryGet<LensDistortion>(out lensdistortion))
+    {
+        killable.OnHealthChange.AddListener(OnHealthChange);
+    }
+
+    if(volume.profile.TryGet<Bloom>(out bloom))
+    {
+        killable.OnHealthChange.AddListener(OnHealthChange);
+    }
+
+   }
+
+
+   void OnHealthChange()
+   {
+    UpdateVisualEffect(killable.currentHP);
+   }
+
+
+   void UpdateVisualEffect(float health)
+   {
+
+    float intensityController = Mathf.Clamp01(1.0f - (health / 100.0f));
+    float negativeintensityController = Mathf.Clamp01(0.0f - (health / 100.0f));
+    float bloomController = Mathf.Clamp01(10.0f - (health / 10.0f));
+
+
+    if(vignetteToggle)
+    {
+        vignette.intensity.value = intensityController;
+    }
+    else
+    {
+        vignette.intensity.value = 0.0f;
```

Comment:
with the implementation I suggested above you dont set the values to 0 because we want some of the post processing